code/game/turfs/open/floor.dm

```diff
@@ -28,22 +28,29 @@
 
 
 /turf/open/floor/Initialize(mapload)
-	if (!broken_states)
-		broken_states = string_list(list("damaged1", "damaged2", "damaged3", "damaged4", "damaged5"))
+	. = ..()
+	if (broken_states)
+		stack_trace("broken_states defined at the object level for [type], move it to setup_broken_states()")
+	else
+		broken_states = string_list(setup_broken_states())
+	if (burnt_states)
+		stack_trace("burnt_states defined at the object level for [type], move it to setup_burnt_states()")
 	else
-		broken_states = string_list(broken_states)
-	if(burnt_states)
-		burnt_states = string_list(burnt_states)
+		burnt_states = string_list(setup_burnt_states())
```

Comment:
```suggestion
		var/list/new_burnt_states = setup_burnt_states()
		if(new_burnt_states)
			burnt_states = string_list(new_burnt_states)
```
This one was fine to keep. After all, there's no guarantee there will always be `burnt_states`